plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/transition/MapboxViewportTransitionFactory.kt

```diff
@@ -0,0 +1,278 @@
+package com.mapbox.maps.plugin.viewport.transition
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.view.animation.Interpolator
+import androidx.core.view.animation.PathInterpolatorCompat
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.plugin.animation.CameraAnimatorOptions.Companion.cameraAnimatorOptions
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.viewport.ViewportPluginImpl.Companion.VIEWPORT_CAMERA_OWNER
+import kotlin.math.abs
+
+/**
+ * Helper class that provides default implementation of [ViewportTransition] generators.
+ */
+internal class MapboxViewportTransitionFactory(
+  mapDelegateProvider: MapDelegateProvider
+) {
+  private val cameraManager = mapDelegateProvider.mapCameraManagerDelegate
+  private val cameraPlugin = mapDelegateProvider.mapPluginProviderDelegate.camera
+
+  fun transitionFromLowZoomToHighZoom(
+    cameraOptions: CameraOptions,
+    transitionMaxDurationMs: Long
+  ): AnimatorSet {
+    val animators = mutableListOf<Animator>()
```

Comment:
I remember we were talking about it - can we use `CameraAnimator`s here?

Reply:
imo, if we don't specifically use the properties defined in CameraAnimator, the generic Animator would be more clean, compared with `mutableListOf<CameraAnimator<*>>`

Reply:
maybe `ValueAnimator` then?

Reply:
I can note it down and try to address it after rc, although I also don't see much added value, as we only use this animators list to construct AnimatorSet and call `AnimatorSet.playTogether(Collection<Animator> items)`.

Reply:
let's leave it current way, I don't have that strong opinion about it, was more curious.